src/ui/modules/EconomicEventManager/index.tsx

```diff
@@ -69,35 +48,29 @@ export const EconomicEventManager: React.FC<EconomicEventManagerProps> = ({
     hasNumericalValue: ''
   });
 
-  const [providerLst, setProviderLst] = React.useState<any>([]);
-  const [receiverLst, setReceiverLst] = React.useState<any>([]);
+  const [providerArr, setProviderArr] = React.useState<any>([]);
+  const [receiverArr, setReceiverArr] = React.useState<any>([]);
   const [unitLst, setUnitLst] = React.useState<any>([]);
-  // const [searchValue, setSearchValue] = React.useState<any>({
-  //   action: '',
-  //   provider: '',
-  //   receiver: '',
-  //   hasUnit: '',
-  // })
 
   React.useEffect(() => {
-    if (economicEvents.data?.length) {
-      const providers = economicEvents.data.map(el => ({
-        id: el.provider.id,
-        label: el.provider.name
-      }));
-      const receivers = economicEvents.data.map(el => ({
-        id: el.receiver.id,
-        label: el.receiver.name
-      }));
-
-      setProviderLst(providers);
-      setReceiverLst(receivers);
-    }
-  }, [economicEvents]);
+    // @ts-ignore
+    const providers = providerList?.map((el: { id: string; name: string }) => ({
```

Comment:
Maybe it's time to create a util function for this sort of mapping, repetition of stuff inn `CreateResourcePanel/index.tsx`. Same goes for the props typing, maybe there's a chance to reuse?